### ISSUE TITLE
Generation fails assuming next xjc argument is an argument to the plugin

### ISSUE THREAD

**Issue by dadrus** [mklemm/jaxb2-rich-contract-plugin#25]
Hi Mirko,

Starting with the version of 1.16.0 of the plugin I can't generate classes out of a wsdl definition. The generation fails with the following output:

```
...
[INFO] jaxws:wsimport args: [-keep, -s, C:\Users\drusdi1\workspace\ias-api-test\target\generated-sources\java, -d, C:\Users\drusdi1\workspace\ias-api-test\target\classes, -encoding, UTF-8, -extension, -Xnocompile, -B-Xcommons-lang, -B-Xfluent-builder, -B-Ximmutable, -b, C:\Users\drusdi1\workspace\ias-api-test\src\main\xjb\jaxb-binding.xml, -b, C:\Users\drusdi1\workspace\ias-api-test\src\main\xjb\jaxws-binding.xml, "file:/C:/Users/drusdi1/workspace/ias-api-test/src/main/resources/META-INF/wsdl/administration.wsdl"]
fluent-builder: Argument "-B-Ximmutable" not supported by "fluent-builder" plugin.


Usage: wsimport [options] <WSDL_URI>

where [options] include:
  -b <path>                 specify jaxws/jaxb binding files or additional schemas
                            (Each <path> must have its own -b)
  -B<jaxbOption>            Pass this option to JAXB schema compiler
  -catalog <file>           specify catalog file to resolve external entity references
                            supports TR9401, XCatalog, and OASIS XML Catalog format.
  -d <directory>            specify where to place generated output files
  -encoding <encoding>      specify character encoding used by source files
  -extension                allow vendor extensions - functionality not specified
                            by the specification.  Use of extensions may
                            result in applications that are not portable or
                            may not interoperate with other implementations
  -help                     display help
  -httpproxy:<proxy>        set a HTTP proxy. Format is [user[:password]@]proxyHost:proxyPort
                            (port defaults to 8080)
  -keep                     keep generated files
  -p <pkg>                  specifies the target package
  -quiet                    suppress wsimport output
  -s <directory>            specify where to place generated source files
  -target <version>         generate code as per the given JAXWS spec version
                            Defaults to 2.2, Accepted values are 2.0, 2.1 and 2.2
                            e.g. 2.0 will generate compliant code for JAXWS 2.0 spec
  -verbose                  output messages about what the compiler is doing
  -version                  print version information
  -fullversion              print full version information
  -wsdllocation <location>  @WebServiceClient.wsdlLocation value
  -clientjar <jarfile>      Creates the jar file of the generated artifacts along with the
                            WSDL metadata required for invoking the web service.
  -generateJWS              generate stubbed JWS implementation file
  -implDestDir <directory>  specify where to generate JWS implementation file
  -implServiceName <name>   local portion of service name for generated JWS implementation
  -implPortName <name>      local portion of port name for generated JWS implementation

Extensions:
  -XadditionalHeaders              map headers not bound to request or response message to 
                                   Java method parameters
  -Xauthfile                       file to carry authorization information in the format 
                                   http://username:password@example.org/stock?wsdl
  -Xdebug                          print debug information
  -Xno-addressing-databinding      enable binding of W3C EndpointReferenceType to Java
  -Xnocompile                      do not compile generated Java files
  -XdisableAuthenticator           disable Authenticator used by JAX-WS RI,
                                   -Xauthfile option will be ignored if set
  -XdisableSSLHostnameVerification disable the SSL Hostname verification while fetching
                                   wsdls

Examples:
  wsimport stock.wsdl -b stock.xml -b stock.xjb
  wsimport -d generated http://example.org/stock?wsdl

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.893 s
[INFO] Finished at: 2016-08-17T17:10:32+02:00
[INFO] Final Memory: 9M/150M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jvnet.jax-ws-commons:jaxws-maven-plugin:2.3:wsimport (default) on project ias-api-test: Mojo failed - check output -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
...
```

The corresponding pom fragment looks like this:

``` .xml
<plugin>
    <groupId>org.jvnet.jax-ws-commons</groupId>
    <artifactId>jaxws-maven-plugin</artifactId>
    <version>2.3</version>

    <dependencies>
        <dependency>
            <groupId>net.codesup.util</groupId>
            <artifactId>jaxb2-rich-contract-plugin</artifactId>
            <version>1.16.0</version>
        </dependency>
        <dependency>
            <groupId>org.jvnet.jaxb2_commons</groupId>
            <artifactId>jaxb2-commons-lang</artifactId>
            <version>2.4</version>
        </dependency>
    </dependencies>

    <executions>
        <execution>
            <goals>
                <goal>wsimport</goal>
            </goals>
            <configuration>
                <wsdlDirectory>${project.basedir}/src/main/resources/META-INF/wsdl</wsdlDirectory>
                <wsdlLocation>http://www.test.com?wsdl</wsdlLocation>
                <keep>true</keep>
                <sourceDestDir>${project.basedir}/target/generated-sources/java</sourceDestDir>
                <staleFile>${project.build.directory}/jaxws/stale/wsdl.generation.done</staleFile>
                <bindingDirectory>${project.basedir}/src/main/xjb</bindingDirectory>
                <catalog>${project.basedir}/src/main/resources/META-INF/jax-ws-catalog.xml</catalog>
                <extension>true</extension>
                <xjcArgs>
                    <xjcArg>-Xcommons-lang</xjcArg>
                    <xjcArg>-Xfluent-builder</xjcArg>
                    <xjcArg>-Ximmutable</xjcArg>
                </xjcArgs>
            </configuration>
        </execution>
    </executions>
</plugin>
```

If I remove the last `<xjcArg>` argument, everything is fine. But I need the immutability feature.

What am I doing wrong?

Regards,
Dimitrij

**Comment by mklemm** [comment]
There seems to be something wrong in the way the argument line is parsed. The jaxws plugin seems to add a "-B" before every option, and that seems to cause problems. Maybe you could try a different order of the <xjcArg> options? I'll have a look at the code to see if it is a bug that can be fixed in the jaxb2-rich-contract plugin.

**Comment by dadrus** [comment]
Before v1.16.0 of the plugin, everything is working as expected, without the need of reordering the arguments. Btw. reordering does not help.

**Comment by mklemm** [comment]
Have you tried with version 1.17.0?

**Comment by dadrus** [comment]
Sorry for the belated answer. Yes. 1.16.0 and 1.17.0 both lead to the same error. The error does not appear with 1.15.0 and below.

**Comment by unclealex72** [comment]
I've submitted the following [pull request](https://github.com/mklemm/jaxb-plugin-lib/pull/1) for this.

**Comment by martijn-van-tilburg** [comment]
The issue seems not to be solved for my case, version 1.15.0 works though:

`[INFO] jaxws:wsimport args: [-keep, -s, C:\repos\my-project\webservice\webservice-shared\target\generated-sources\wsimport, -d,
 C:\repos\my-project\webservice\webservice-shared\target\classes, -verbose, -encoding, UTF-8, -Xnocompile, -p, 
 org.myproject.generated.myservice, -Xdebug, -B-Xfluent-builder, 
 "file:/C:/repos/my-project/webservice/webservice-shared/src/main/resources/myservice.wsdl"]`

**Comment by gavenkoa** [comment]
@martijn-van-tilburg It is only partially solved. Look into workaround in:

https://github.com/mklemm/jaxb2-rich-contract-plugin/issues/47
  Gradle + Ant wsimport task + wsdllocation + fluent API fail with: "Argument
  "-wsdllocation" not supported by "fluent-builder" plugin".
https://github.com/mklemm/jaxb-plugin-lib/issues/2
  AbstractPlugin.parseArgument is greedy for options. This breaks wsimport utility.

**Comment by mklemm** [comment]
Issue was fixed in 4.0.0 which requires prefixing all plugin command line options with the plugin name. See README.md
